### ISSUE TITLE
Worldwide rankings

### ISSUE THREAD

**Issue by SolalPirelli** [CSStipendRankings/CSStipendRankings#52]
Hi there! I'm a CS PhD student at EPFL in Switzerland so I'd love to add more non-US data to this. Hope this isn't too much of a wall of text;
if a discussion over everything is too much, I could open sub-issues, but I didn't want to do so at first since that might be considered spamming.

I'm happy to work on any or all of the points below, but I'd like to get some consensus before sinking time into it :-)

This issue tracks necessary work to include universities worldwide instead of only in the US.
This may not be exhaustive as different countries may have unexpected requirements, but I've tried my best using information I know about various places.

The goal remains to make departments pay PhD students a living wage; using international comparisons can help further that goal by highlighting deficiencies specific to US academia.
Thus, the following are what I think is necessary for a "worldwide" ranking to make sense.

- **Flip the summer classification** to mark universities who do _not_ pay a full 12 months, as this seems like a US-specific thing.
  One way to do so would be to use a "months of pay" column  if it's not 12, i.e, "9" for no summer pay, and "10.5" for half-pay in summer. The stipend would be scaled if necessary.

- **Drop the explicit pre/post-qual salary distinction** as this is US-specific and anyway the point is people should be paid a living wage, not only if they've passed some examination.
  Perhaps this can be replaced by a "notes" field that includes such information.

- **Include fees in salaries** instead of making them explicit; while they might influence cost of living (e.g., if the university provides health care "subsidies"), they're an accounting trick.
  The concept of "tuition" fees especially seems North American, there's no reason to separate it from the salary.

- **Switch the (default) salary methodology to "minimum possible"**, not what someone may or may not get depending on various factors.
  This is in line with the goal of encouraging universities to pay more, not only to some people in some situations. (Yes, this will make some unis look terrible -- that's the point)
  See also #43 in general.

- **Add salary conditions as a column**, such as how much time has to be dedicated to teaching given the salary, or any other such conditions.
  This would help people make decisions and also make the ranking more fair to universities, since salaries are not all for the same set of tasks.

- **Split data for cost of living** to make collecting data easier since the MIT calculator total uses a specific methodology, but also provides details.
  Obviously, these could still be summed up in the displayed data.
  I suggest splitting into "food", "health", "housing", "infrastructure" (incl. whatever transportation is reasonably necessary), "taxes" (but see #50), "other".

- **Come up with a reasonable methodology for health costs** that would mostly make sense worldwide.
  I believe that pretty much all countries nowadays have some form of mandated health insurance, which may be paid through taxes or not, so at least we can use the concept of a legally-required plan instead of making up requirements.
  I suggest something like "cost of the cheapest legally-required plan for a 24 year old grad student, with subsidies if available, plus cost of a yearly dental checkup if not included".
  This may make some plans look worse than they are due to copays, so perhaps the metric could also include "payable cost for an average knee surgery" or some other representative cost that one would reasonably expect might happen.

- **Add an explicit "source" column to each row** to indicate where the data comes from. (This would be a lot of text, it can be hidden behind a drop-down or hover)
  This is anyway required since the MIT living wage calculator is US-only, but could also be used to indicate other sources such as the salary itself.
  It would also include methodological details such as what was chosen for transportation (walking? public transportation? a car?) if needed.

- **Come up with a methodology for currency conversion**, since showing different currencies is too hard to parse, yet showing the exact conversion at the second the page was loaded risks making the ranking too volatile.
  Since living wages etc. are computed yearly (AFAIK), one way would be to use the currency conversion on January 1st of the current year?

- **Make the "after fees and living" amount relative**, since it makes little sense even with $-converted amounts.
  Perhaps use some index like the Big Mac Index, i.e., how many Big Macs can you buy with the money left after living expenses?

**Comment by pyjhzwh** [comment]
Hello Solal,

I appreciate your thoughts and your plan to include non-US universities. Here are some of my thoughts.

- I agree that we could use a 9-month, 12-month salary rather than a "summer" label. It could be more worldwide understandable.
- But we may need to keep the pre/post-qual salary. Because in some universities, first- and second-year Ph.D. students usually take lots of courses to meet the course requirements, the tuition fee is high. However, some universities will raise their salary when they pass the prelim/qualification and become Ph.D. candidates, given that they do not actually take courses. If it is US-specific, just make pre/post-qual salary the same number.
- I still think the fees should be separate from the salaries. The fees are not tuition fees (they would not be that cheap lol). It may be international fees or other fees that Ph.D. students are charged by universities separately from their guaranteed salary. We could be more clear about the definition of "fees" here.
- I also agree that we should set the salary as the guaranteed funding ("minimum possible") from the university.
- It is a good idea to add "salary conditions". When people click the salary, the details of the "salary conditions" can be expanded. It may be too much information if this is added as another column.
- Regarding making "after fees and living" amount relative, #33

**Comment by SolalPirelli** [comment]
Thanks! I'll add that issue ref to the original post.

Re: fees, do you (or anyone else ofc) have concrete examples of such fees in the US / elsewhere?

**Comment by mjc0608** [comment]
>fees, do you (or anyone else ofc) have concrete examples of such fees in the US / elsewhere?

This includes things like international fees, student government fees, etc. Some universities do charge some weird fees that do not make sense to me. You may find some of them in the issue history. Also, health insurance is included in the fees, because otherwise it would be unfair for universities that provide free health insurance.

>Flip the summer classification

This is a good idea but requires some implementation efforts. It will not happen at least in the next two weeks. However, we are willing to merge patches submitted by volunteers for this.

>non-US data

I believe @TonyZhangND is working on adding support for other countries. However, this requires some fundamental code change and may take a while.

>Worldwide rankings

This does not make sense to me. For example, in China, the typical CS PhD salary is $800 a month, but with that money, your life will be really comfortable, since you only need to spend $300 for dinning and the housing is basically free.

>Switch the (default) salary methodology to "minimum possible"

We will add a switch in the future. Maybe we will use what most people get as a default.

>Add salary conditions as a column; Split data for cost of living; Come up with a reasonable methodology for health costs; Come up with a methodology for currency conversion

These will not happen in the next two weeks unless contributed by a volunteer.

>Add an explicit "source" column to each row

We will consider it.

>Make the "after fees and living" amount relative

We are considering alternative metrics. However, it's hard to find the data and it's mostly out of the scope of our knowledge. Any pull requests are welcome.

**Comment by TonyZhangND** [comment]
> non-US data

I didn't get very far before my work got broken by another commit, so feel free to have a stab if you want @SolalPirelli

**Comment by SolalPirelli** [comment]
> This includes things like international fees, student government fees, etc. Some universities do charge some weird fees that do not make sense to me. You may find some of them in the issue history. Also, health insurance is included in the fees, because otherwise it would be unfair for universities that provide free health insurance.

If a uni provides health insurance, it'll be reflected in a lower cost of living, but fundamentally it's still an accounting trick as they could add that money to your pay instead and have you pay for your own insurance, no?

If some US state decided to provide state-run health care, it would also be unfair to universities not in that state, but we'd presumably classify it as lowering of the cost of living.

**Comment by sharad1126** [comment]
Hi, I am a Ph.D. student at University College London (UCL), London, UK, and would be happy to contribute to the non-US data.

**Comment by mjc0608** [comment]
I think for now you can add the data to `stipend-eu.csv`? Displaying non-US data requires change the UI and some js code, and we also need someone to maintain that part (e.g., publishing surveys when data from different sources shows conflicts). Let me know if you want to help.

**Comment by mjc0608** [comment]
A world-wide ranking is unlikely to happen for now. I'm closing this issue.
